tests/test_frame.py

```diff
@@ -3448,41 +3449,72 @@ def test_groupby_apply() -> None:
     def sum_mean(x: pd.DataFrame) -> float:
         return x.sum().mean()
 
-    with pytest_warns_bounded(
-        DeprecationWarning,
-        "DataFrameGroupBy.apply operated on the grouping columns.",
-        upper="2.99",
+    with (
+        pytest_warns_bounded(
+            DeprecationWarning,
+            "DataFrameGroupBy.apply operated on the grouping columns.",
+            upper="2.2.99",
+        ),
+        pytest_warns_bounded(
```

Comment:
Quite annoying that they changed the type of warning that was raised but guess we don't have too much choice than doubling the context manager to grab the warnings.

Reply:
We are testing with 2.3, so you only need to test for `FutureWarning` with `upper="2.3.99"`